### ISSUE TITLE
Probe to check whether the libraries are deprecated or not in the plugin.

### ISSUE THREAD

**Issue by AayushSaini101** [jenkins-infra/plugin-health-scoring#470]
### Description

* Upgrade JUnit 3 tests to JUnit 4
* Replace joda-time with Java 11 time

These are the need of changes required as per plugin modernisation

**Comment by krisstern** [comment]
Sounds like a cool idea to have as a probe, but to be up-to-date I would say maybe JUnit 5 instead of 4 even? Maybe a good idea to do some research on what this probe ideally should cover, as I suspect there is a lot more.

**Comment by alecharp** [comment]
The probes are non judgemental. The probes (one for each) should be to get the versions of known libraries.
Then, we can have a scoring (PluginMaintenance?) which would evaluate those results and determine if the plugin is using up to date or not libraries.

However, there is no recommendation to use JUnit 5 over JUnit 4 in the community.
Also, some plugins are using libraries which are using Joda Time. There is nothing wrong in using Joda time rather than JDK classes. We even just added the Joda Time API plugin (https://github.com/jenkinsci/joda-time-api-plugin). I'm willing to agree and say JDK >>> Joda Time but there is no clear statement that we should move away from Joda Time.

**Comment by AayushSaini101** [comment]
> The probes are non judgemental. The probes (one for each) should be to get the versions of known libraries. Then, we can have a scoring (PluginMaintenance?) which would evaluate those results and determine if the plugin is using up to date or not libraries.
> 
> However, there is no recommendation to use JUnit 5 over JUnit 4 in the community. Also, some plugins are using libraries which are using Joda Time. There is nothing wrong in using Joda time rather than JDK classes. We even just added the Joda Time API plugin (https://github.com/jenkinsci/joda-time-api-plugin). I'm willing to agree and say JDK >>> Joda Time but there is no clear statement that we should move away from Joda Time.

I agree with @alecharp It is good to have a probe for each to get the version of known libraries. Regarding the migration of the version from Upgrade JUnit 3 tests to JUnit 4 and Replace joda-time with Java 11 time it is mentioned in the Contribute to Open Source guidelines by Jenkins
